### ISSUE TITLE
Move location where to include defines_custom.inc.php 

### ISSUE THREAD

**Issue by amtriorix** [PrestaShop/PrestaShop#12724]
To debug PS you can alter config/defines.inc.php, however config.inc.php does provide
a mechanishm to use config/defines_custom.inc.php as well that is a more elegant solution.

create config/defines_custom.inc.php
```
<?php
//
//
//
define('_PS_MODE_DEV_', true);

```

This works for a running PS however this file is not properly read using i
install/index.php using the call init.php.

We see,  defines.inc.php however, is included by init.php, so.

Proposed solution: 

Move location where defines_custom.inc.php is included.
--------------------------------------------------
Old call  : config.inc.php

remove:
```
/* Custom defines made by users */
if (is_file($currentDir.'/defines_custom.inc.php')) {
    include_once($currentDir.'/defines_custom.inc.php');
}
```
----------------------------------------------------
New call: defines.inc.php
```
$currentDir = dirname(__FILE__);
/* Custom defines made by users */
if (is_file($currentDir.'/defines_custom.inc.php')) {
    include_once($currentDir.'/defines_custom.inc.php');
}
/* Debug only */
if (!defined('_PS_MODE_DEV_')) {
define('_PS_MODE_DEV_', false);
}
...
// remove line 45 $currentDir = dirname(__FILE__);

```

Not tested.

**Comment by khouloudbelguith** [comment]
Hi @amtriorix,

Have you the same issue with this ticket: https://github.com/PrestaShop/PrestaShop/issues/11234?
Thanks!

**Comment by amtriorix** [comment]
No, this is to move the location when and where to include properly the custom defines,
so it also works doing install/index.php  Now this file is not read by install/index.php (using install/init.php)

To solve it properly, I propose to call the include in config/defines.inc.php

**Comment by khouloudbelguith** [comment]
@amtriorix, thanks for these clarifications.
Checked with PS1.7.5.1 & PS1.7.4.4.
I added the new file 

config/defines_custom.inc.php

```
<?php

define('_PS_MODE_DEV_', true);
```
I try to navigate to the BO => ok => The debug mode is enabled
When I try to re-install this shop => THe debug mode is disabled => so the `defines_custom.inc.php` is not treated.

@amtriorix, would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop
Thank you!

**Comment by marionf** [comment]
@khouloudbelguith Could you try to reproduce it with 1.7.6.x ?

**Comment by khouloudbelguith** [comment]
Hi @marionf,

I have the same issue with PS1.7.6.0beta1.
I added the new file

`config/defines_custom.inc.php
`
```
<?php

define('_PS_MODE_DEV_', true);
```
When I try to re-install this shop => The debug mode is disabled
![image](https://user-images.githubusercontent.com/16067358/59193026-3735f780-8b7c-11e9-86cb-d2164d241d25.png)

Thanks!
